libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -397,21 +396,32 @@ void AP_Mount_Siyi::process_packet()
         break;
 
     case SiyiCommandId::ACQUIRE_GIMBAL_CONFIG_INFO: {
-        // update gimbal's mounting direction
-        if (_parsed_msg.data_bytes_received > 5) {
-            _gimbal_mounting_dir = (_msg_buff[_msg_buff_data_start+5] == 2) ? GimbalMountingDirection::UPSIDE_DOWN : GimbalMountingDirection::NORMAL;
+        const RecordingStatus prev_record_status = _config_info.record_status;
+
+        // Update Gimbal Config Information
+        size_t config_sz = MAX(_parsed_msg.data_bytes_received, sizeof(_config_info));
```

Comment:
Hi, I'm not sure this is safe because if the gimbal sends a large message then it could overrun the _config_info structure.  Maybe it should be MIN?

Reply:
Yep, that's a bug. Fixed.